provisioning/WFLY-18671_Galleon_tooling_upward_compat.adoc

```diff
@@ -0,0 +1,219 @@
+
+=== Impact on tooling
+
+This API is to be used by Provisioning tooling. For example:
+
+* Galleon CLI and Maven plugin
+* WildFly Maven plugin
+* WildFly bootable JAR Maven plugin
+* Prospero CLI
+* WildFly Glow
+
+The impact is at the tooling implementation level, no visible changes at the tooling user level is expected. For example, 
+the provisioning based on Galleon Maven plugin that occurs in WildFly repository is untouched.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-18671[WFLY-18671]
+
+=== Related Issues
+
+* 
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* Tested by Eng.
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/galleon/[Galleon]
+
+* https://github.com/wildfly/galleon-plugins/[Galleon Plugins]
+
+* https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin]
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin[WildFly Bootable JAR]
+
+* https://github.com/wildfly-extras/prospero[Prospero]
+
+* https://github.com/wildfly/wildfly-glow[WildFly Glow]
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Implementation notes
+
+==== Tooling API
+
+The API is composed of 3 parts. Being 3 parts is an implementation detail, tooling depend directly on the Galleon API only.
+
+* Galleon maven universe: The resolution of Galleon feature-packs using Maven.
+
+* Galleon common API: this API is shared between Galleon core and the Galleon API.
+
+* Galleon API: Provisioning + feature-pack resolutions. It depends on the Galleon common API and Galleon maven universe.
+
+The maven coordinates of the Galleon API: `org.jboss.galleon:galleon-api", coordinates of the common api: `org.jboss.galleon:galleon-common-api`, 
+coordinates of the maven universe api: `org.jboss.galleon:galleon-maven-universe`.
+
+==== No Galleon core classes in the tooling classpath
+
+The Galleon core classes must not be present in the tooling classpath. Dynamic classloader is to be used to resolve Galleon core classes.
+
+==== Resolution of the Galleon core library
+
+At provisioning time, the new API resolves the feature-pack(s) to use and retrieve the Galleon core version advertised by the feature-pack(s).
+When multiple galleon feature-packs are used, the greater version is used. If a discovered version is smaller than the API version, the API version is used.
+If no version is discovered (feature-pack built with an older version of Galleon), the version of the API in use is used. 
```

Comment:
> the version of the API in use is used

Just to confirm - the Galleon Core with the same version as the API is used right?